### PR TITLE
p.v sub-patent duration is immunity dependent.

### PR DESCRIPTION
**Pull request by RJSheppard**
antiparasite immunity function shapes the duration of subpatent infection/rate of recovery from subpatent infection. These values are now assigned during the variable creation and calculation of recovery rates.

In previous iterations I have kept ID in the vivax model, where it acts as antiparasite immunity. For clarity of thought, I've now added IAA and IAM antiparasite immunities, including their decay, boosting, maternal immunity and reset at death.

We must also adjust the prevalence rendering, which for p.f is based on detectability immunity, and for p.v is explicit in the human states. p_detect_lm is also inappropriate for p.v as the lm_detection is explicit.

du, and ID parameters removed for vivax.

Tests are added for du rates and for prevalence rendering.

### CODE REVIEW

File: R/render.R [hunk 1]
```diff
@@ -26,12 +27,17 @@ create_prevelance_renderer <- function(
...
-    asymptomatic_detected <- bitset_at(asymptomatic, bernoulli_multi_p(prob))
+    
+    if(parameters$parasite == "falciparum"){
+      prob <- probability_of_detection(
+        get_age(birth$get_values(asymptomatic), timestep),
+        immunity$get_values(asymptomatic),
+        parameters
+      )
+      asymptomatic_detected <- bitset_at(asymptomatic, bernoulli_multi_p(prob))
+    } else {
```
Comment (pwinskill) [inline R/render.R:38]: You seem to mostly have `elseif (parameters$parasite == "vivax")` - is there a reason this is different?

File: R/disease_progression.R [hunk 2]
```diff
@@ -6,11 +6,22 @@
...
-    recovery_outcome$set_rates(variables$recovery_rates$get_values())
+    recovery_rates <- variables$recovery_rates$get_values()
+    if (parameters$parasite == "vivax"){
+      # p.v subpatent recovery is immunity-dependent
+      recovery_rates[variables$state$get_index_of("U")$to_vector()] <-
+        1/anti_parasite_immunity(
+          parameters$dpcr_min, parameters$dpcr_max, parameters$apcr50, parameters$kpcr,
+          variables$iaa$get_values(index = variables$state$get_index_of("U")),
+          variables$iam$get_values(index = variables$state$get_index_of("U"))
+        )
```
Comment (pwinskill) [inline R/disease_progression.R:22]: Hard to parse like this, I think we can make it more efficient too (untested suggestion).
```suggestion
  # p.v subpatent recovery is immunity-dependent
  
  U_index <- variables$state$get_index_of("U")
  
  recovery_rates[U_index$to_vector()] <-
    1 / anti_parasite_immunity(
      min = parameters$dpcr_min,
      max = parameters$dpcr_max,
      a50 = parameters$apcr50,,
      k = parameters$kpcr,
      iaa = variables$iaa$get_values(index = U_index),
      iam =variables$iam$get_values(index = U_index)
    )
```

**Review by pwinskill — CHANGES_REQUESTED** [review]

**Review by pwinskill — APPROVED** [review]

**Review by giovannic — APPROVED** [review]
Thank you!
